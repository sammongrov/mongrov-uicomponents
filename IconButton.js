import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import { styles } from 'react-native-theme';
import PropTypes from 'prop-types';
import Icon from './Icon';

export default class IconButton extends Component {
  render() {
    const { containerStyle, icon, iconStyle, onPress } = this.props;
    return (
      <TouchableOpacity onPress={onPress} style={[styles.authIBtncontainer, containerStyle]}>
        <Icon active name={icon} style={[styles.authIBtnIcon, iconStyle]} />
      </TouchableOpacity>
    );
  }
}

IconButton.propTypes = {
  containerStyle: PropTypes.number.isRequired,
  iconStyle: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

IconButton.defaultProps = {};
