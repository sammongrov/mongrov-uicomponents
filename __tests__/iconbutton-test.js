import React from 'react';
import { Alert } from 'react-native';

import renderer from 'react-test-renderer';
import IconButton from '../IconButton';

const onPress = () => Alert.alert('Accepted');

it('IconButton renders correctly', () => {
  const tree = renderer.create(<IconButton onPress={onPress} />).toJSON();
  expect(tree).toMatchSnapshot();
});
