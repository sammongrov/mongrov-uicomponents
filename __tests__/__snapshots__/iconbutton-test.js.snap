// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    active={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          undefined,
          undefined,
        ],
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
